dataikuapi/dss/ml.py

```diff
@@ -862,22 +875,36 @@ def get_prediction_info(self):
             }
 
 
-class DSSSubpopulationModality(DSSInternalDict):
+class DSSSubpopulationModality(object):
     """
     Object to read details of a subpopulation analysis modality
 
     Do not create this object directly, use :meth:`DSSSubpopulationAnalysis.get_modality_data(definition)` instead
     """
 
     def __init__(self, feature_name, computed_as_type, data, prediction_type):
-        super(DSSSubpopulationModality, self).__init__(data)
+        if data is None:
+            self._internal_dict = dict()
+        else:
+            self._internal_dict = data
 
         self.prediction_type = prediction_type
         if computed_as_type == "CATEGORY":
             self.definition = DSSSubpopulationCategoryModalityDefinition(feature_name, data)
         elif computed_as_type == "NUMERIC":
             self.definition = DSSSubpopulationNumericModalityDefinition(feature_name, data)
-    
+
+    def get_raw(self):
+        """
+        Gets the raw dictionary of the actual data
+
+        :rtype: dict
+        """
+        return self._internal_dict
+
+    def __repr__(self):
+        return self.__class__.__name__ + "(" + self._internal_dict.__repr__() + ")"
```

Comment:
Should return the modality (category value or numerical bin) rather than the full data